src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,111 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_ARRAYLIST_SIZE = 10;
+    private int size;
+
+    private T[] array;
+
+    public ArrayList() {
+        array = (T[]) new Object[DEFAULT_ARRAYLIST_SIZE];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == array.length) {
+            growArray();
+        }
+        array[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == size) {
+            add(value);
+            return;
+        }
+        checkIndex(index);
+        if (size == array.length) {
+            growArray();
+        }
+        System.arraycopy(array, index, array, index + 1,array.length - index - 1);
+        array[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index == size) {
+            throw new ArrayListIndexOutOfBoundsException("Index out of array list range");
+        }
+        checkIndex(index);
+        return array[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        if (index == size) {
+            throw new ArrayListIndexOutOfBoundsException("Index out of array list range");
+        }
+        checkIndex(index);
+        array[index] = value;
+        if (index > size) {
+            size = index;
+        }
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        if (index == size) {
+            throw new ArrayListIndexOutOfBoundsException("Index out of array list range");
+        }
+        checkIndex(index);
+        T result = array[index];
+        System.arraycopy(array,index + 1, array, index,array.length - index - 1);
+        size--;
+        return result;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        for (int i = 0; i < size; i++) {
+            if (array[i] == element || element != null && element.equals(array[i])) {
+                return remove(i);
+            }
+        }
+        throw new NoSuchElementException();
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void growArray() {
+        T[] tempArray = (T[])(new Object[array.length + (array.length >> 1)]);
+        System.arraycopy(array, 0, tempArray,0, array.length);
+        array = (T[]) new Object[array.length + (array.length >> 1)];
+        array = tempArray;
```

Comment:
not implemented like it should be
![image](https://user-images.githubusercontent.com/81781479/185743909-0abb3ef4-011d-4530-a872-6030ff27cf1e.png)

Reply:
tempArray = (T[]) new Object[array.length * 1.5]

Reply:
![image](https://user-images.githubusercontent.com/81781479/185743962-d250dbbe-6974-424a-890a-f862603ba345.png)
on 102 line you create an array, with all elements null? you don't need this array